app/javascript/views/ActivateAccount.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Fragment } from 'react';
+import { inject, observer } from 'mobx-react';
+import SessionStore, { RequestState } from '../stores/session_store';
+import { Header, Form, Message, Dimmer, Loader } from 'semantic-ui-react';
+import { RouterStore } from 'mobx-react-router';
+import DbSqlIcon from '../shared/DbSqlIcon';
+import { RouteComponentProps } from 'react-router';
+import { resetPassword as resetPasswordCall, activateAccount } from '../api/user';
+
+interface MatchParams {
+  id: string;
+}
+interface ResetPasswordProps extends RouteComponentProps<MatchParams> { }
+
+interface InjectedProps extends ResetPasswordProps {
+  sessionStore: SessionStore;
+  routerStore: RouterStore;
+}
+
+@inject('sessionStore', 'routerStore')
+@observer
+export default class ActivateAccount extends React.Component<ResetPasswordProps> {
+  state = {
+    requestState: RequestState.None,
+    errorMsg: ''
+  };
+
+  componentDidMount() {
+    this.setState({ requestState: RequestState.Waiting });
+    activateAccount(this.id, this.activationToken).then(() => {
+      this.setState({ requestState: RequestState.Success });
+    }).catch((error) => {
+      this.setState({
+        requestState: RequestState.Error,
+        errorMsg: error.response.data.error || 'Unexpected server error'
+      });
+    });
+  }
+
+  get injected() {
+    return this.props as InjectedProps;
```

Comment:
why not assigning the return values of those getters inside the render method to a const variable?